das/daser.go

```diff
@@ -143,6 +154,14 @@ func (d *DASer) sample(ctx context.Context, sub header.Subscription, checkpoint
 			if err == context.Canceled {
 				return
 			}
+			var byzantineErr *ipld.ErrByzantine
```

Comment:
This function can also touch parts of the code not introduced in this PR, so let's make it after we merge this. I think you will be in the best position to see how this can DRYied out when you start to do DASState.